### PR TITLE
Fields

### PR DESCRIPTION
**Pull request by gertjanvanzwieten**

### CODE REVIEW

File: nutils/expression_v2.py [hunk 1]
```diff
@@ -709,6 +709,26 @@ def define_for(self, __name: str, *, gradient: Optional[str] = None, curl: Optio
                 raise ValueError('Cannot define the jacobian {!r}: dimension is negative.'.format(jacobian))
             setattr(self, jacobian, function.jacobian(geom, numpy.size(geom) - i))
 
+    def add_field(self, __names: tuple, __bases: tuple = (), shape: Tuple[int, ...] = (), dtype: function.DType = float):
```
Comment (joostvanzwieten) [inline nutils/expression_v2.py:712]: The annotation for `names` should be `Union[str, Sequence[str]]`. The annotation for `bases` is incorrect as well. Perhaps it's best to remove it.

File: examples/cylinderflow.py [hunk 2]
```diff
@@ -58,32 +58,33 @@ def main(nelems: int, degree: int, reynolds: float, rotation: float, timestep: f
...
         domain.basis('spline', degree=(degree, degree-1), removedofs=((0,), None)),
         domain.basis('spline', degree=(degree-1, degree))]), J.T) / detJ
-    ns.pbasis = domain.basis('spline', degree=degree-1) / detJ
-    ns.u = function.dotarg('u', ns.ubasis)
-    ns.p = function.dotarg('p', ns.pbasis)
+    pbasis = domain.basis('spline', degree=degree-1) / detJ
+    ns.u = function.dotarg('u', ubasis)
+    ns.v = function.dotarg('v', ubasis)
+    ns.p = function.dotarg('p', pbasis)
+    ns.q = function.dotarg('q', pbasis)
```
Comment (joostvanzwieten) [inline examples/cylinderflow.py:68]: I think we can use `ns.add_field` here as well.

**Review by joostvanzwieten — APPROVED** [review]

**Comment by joostvanzwieten** [comment]
Both driven cavity examples have a `function.dotarg` in `postprocess` that can be replaced with a `ns.add_field`.

File: nutils/expression_v2.py (hunk 1 above)
Comment (gertjanvanzwieten) [reply nutils/expression_v2.py]: fixed (and turned bases into varargs)

File: examples/cylinderflow.py (hunk 2 above)
Comment (gertjanvanzwieten) [reply examples/cylinderflow.py]: fixed

**Pull request merged**